astropy/modeling/core.py

```diff
@@ -1961,14 +1965,15 @@ class members that should be added to the generated
         ``standard_broadcasting``, and ``__module__`.  This is currently for
         internal use only.
         """
-
         # Note, currently this only supports binary operators, but could be
         # easily extended to support unary operators (namely '-') if/when
         # needed
         children = []
         for child in (left, right):
             if isinstance(child, (_CompoundModelMeta, _CompoundModel)):
-                children.append(child._tree)
+                children.append(copy.deepcopy(child._tree))
```

Comment:
I think here it would be good to add a comment to explain why you are doing a deep copy